pkg/noderesourcetopology/filter_test.go

```diff
@@ -292,6 +320,38 @@ func TestNodeResourceTopology(t *testing.T) {
 			node:       nodes[1],
 			wantStatus: framework.NewStatus(framework.Unschedulable, fmt.Sprintf("cannot align container: %s", containerName)),
 		},
+		{
+			name: "Burstable QoS requesting CPU (enough on NUMA) and devices, Pod Scope Topology policy; pod fit",
+			pod: makePodByResourceList(&v1.ResourceList{
+				v1.ResourceCPU:  *resource.NewQuantity(2, resource.DecimalSI),
+				nicResourceName: *resource.NewQuantity(5, resource.DecimalSI)}),
+			node:       nodes[2],
+			wantStatus: nil,
+		},
+		{
+			name: "BurstableQoS requesting CPU (not enough on NUMA) and devices, Pod Scope Topology policy; pod fit",
+			pod: makePodByResourceList(&v1.ResourceList{
+				v1.ResourceCPU:  *resource.NewQuantity(20, resource.DecimalSI),
+				nicResourceName: *resource.NewQuantity(5, resource.DecimalSI)}),
+			node:       nodes[2],
+			wantStatus: nil,
+		},
+		{
+			name: "Burstable QoS requesting CPU (enough on NUMA) and devices, Container Scope Topology policy; pod fit",
+			pod: makePodByResourceList(&v1.ResourceList{
+				v1.ResourceCPU:  *resource.NewQuantity(2, resource.DecimalSI),
+				nicResourceName: *resource.NewQuantity(5, resource.DecimalSI)}),
+			node:       nodes[1],
+			wantStatus: nil,
+		},
+		{
+			name: "Burstable QoS requesting CPU (not enough on NUMA) and devices, Container Scope Topology policy; pod fit",
+			pod: makePodByResourceList(&v1.ResourceList{
+				v1.ResourceCPU:  *resource.NewQuantity(4, resource.DecimalSI),
+				nicResourceName: *resource.NewQuantity(5, resource.DecimalSI)}),
+			node:       nodes[1],
+			wantStatus: nil,
+		},
```

Comment:
to avoid introducing regressions: do we have tests covering the case on which we have burstable/best-effort pods requesting cpu/memory/both such that requested quantity is not available on a single NUMA zone? these pods should still be admitted (per explanation in the commit message)

Reply:
Yeah right, I added more test cases to cover these scenarios.

Reply:
> do we have tests covering the case on which we have burstable/best-effort pods requesting cpu/memory/both such that requested quantity is not available on a single NUMA zone

Isn't the described state a bit of a paradox?
I mean if it's a burstable/best-effort pod it means it shares its resources with others (pod/services) i.e. we don't care about performance so much.
Hence, why do we care from which NUMA the resources are coming?

Reply:
> to avoid introducing regressions: do we have tests covering the case on which we have burstable/best-effort pods requesting cpu/memory/both such that requested quantity is not available on a single NUMA zone?

Isn't the described above a bit of a paradox? 
I mean if it's a burstable/best-effort pod it means it shares its resources with other pods/services, i.e. this pod doesn't care about performance so much.
Hence, why the scheduler should care about NUMA alignment in that case?

According to the [docs](https://kubernetes.io/docs/tasks/administer-cluster/topology-manager/#policy-single-numa-node ) we won't end up with `TopologyAffinityErrror` anyway.
and I quote: 
_In the case of the BestEffort pod, the static CPU Manager policy would send back the default topology hint as there is no CPU request and the Device Manager would send back the hints for each of the requested devices._

Reply:
> > to avoid introducing regressions: do we have tests covering the case on which we have burstable/best-effort pods requesting cpu/memory/both such that requested quantity is not available on a single NUMA zone?
> 
> Isn't the described above a bit of a paradox? I mean if it's a burstable/best-effort pod it means it shares its resources with other pods/services, i.e. this pod doesn't care about performance so much. Hence, why the scheduler should care about NUMA alignment in that case?
> 
> According to the [docs](https://kubernetes.io/docs/tasks/administer-cluster/topology-manager/#policy-single-numa-node) we won't end up with `TopologyAffinityErrror` anyway. and I quote: _In the case of the BestEffort pod, the static CPU Manager policy would send back the default topology hint as there is no CPU request and the Device Manager would send back the hints for each of the requested devices._

We need to take into consideration that CPU and memory are handled differently from devices in case of non-gu pods. In case of burstable and best effort pods, CPU and memory is not allocated exclusively and therefore such pods obtain resources from the shared pool. Devices however, are always exclusively allocated (as per the device plugin API design) irrespective of pod QoS. As per your quote above, `In the case of the BestEffort pod, ... Device Manager sends back hints for each requested device` because of which we _should_ check if the requested devices are available on any of the NUMA nodes on a candidate node. Without the check proposed in this PR, the pod could potentially get scheduled on a node for it to be later rejected with `TopologyAffinityErrror` by Topology Manager.

Reply:
> We need to take into consideration that CPU and memory are handled differently from devices in case of non-gu pods. In case of burstable and best effort pods, CPU and memory is not allocated exclusively and therefore such pods obtain resources from the shared pool. Devices however, are always exclusively allocated (as per the device plugin API design) irrespective of pod QoS. As per your quote above, In the case of the BestEffort pod, ... Device Manager sends back hints for each requested device because of which we should check if the requested devices are available on any of the NUMA nodes on a candidate node. Without the check proposed in this PR, the pod could potentially get scheduled on a node for it to be later rejected with TopologyAffinityErrror by Topology Manager.

I think we're basically saying the same thing.
I was talking about adding more cases with non-gu pods that are asking for CPU/memory - This won't cause  ```TopologyAffinityErrror``` what so ever. only wrong device alignment

Reply:
> Just doublechecking: this is in the narrow, but still relevant, case of a pod requesting _only_ devices, all of which should be available on the same NUMA node. Right?

Yep, I have some test cases [here](https://github.com/kubernetes-sigs/scheduler-plugins/pull/425/commits/149ee7de1ff9ba64efe03d840a736c295e74ba6d#diff-3da058c06e05f2fc9bfa2e6f69abd193781cc354b75c44c4660dc7caaeadf38cR235-R261) corresponding to pods requesting only devices all of which should be allocated from the same NUMA node.

Reply:
> I think we're basically saying the same thing. I was talking about adding more cases with non-gu pods that are asking for CPU/memory - This won't cause `TopologyAffinityErrror` what so ever.
Ah I see. Apologies, I misunderstood your comment previously. I already added more test cases for non-gu pods in the second commit. Let me know if you think I am missing any case that should be added. 
>  Please keep me honest here, but with non-gu pod, only wrong device alignment can cause `TopologyAffinityErrror`
Yes, that's right!